### PR TITLE
feat: export organization members

### PR DESCRIPTION
**Pull request by kaleabendrias**
**What does this PR do?**

This PR introduces an endpoint that enables the export of organisation members to a CSV file. The endpoint will generate a CSV file containing details of all members in an organisation, including their ID, name, email, and role.

**How to Test**

**Export Organisation Members to CSV:**

- Use the endpoint `GET /api/v1/organisations/:org_id/members/export` to export members of an organisation by their organisation ID.

**Verify CSV Export:**

- Check if the CSV file is created with the expected headers (ID, Name, Email, Role) and member data.
- Ensure that the file is downloadable and the data matches the organisation members' details in the database.

**Edge Cases**

- Test whether the organisation exists before attempting to export members.
- Test the behaviour of the system when trying to export members from an organisation that does not exist.
- Test the export functionality for an organisation with no members.

**Checklist**

- Implementing the `/api/v1/organisations/:org_id/members/export` endpoint for exporting members to a CSV file.
- Validating the `org_id` parameter to ensure it refers to an existing organisation.
- Using the `csv-writer` module to generate the CSV file.
- Writing tests to validate the CSV export functionality and edge cases.

### CODE REVIEW

**Review by Heba-WebDev — APPROVED** [review]

**Comment by TheCodeGhinux** [comment]
Fix your conflicts

**Comment by kaleabendrias** [comment]
@TheCodeGhinux i resolved it now.

File: src/modules/organisations/organisations.controller.ts [hunk 1]
```diff
@@ -154,4 +156,51 @@ export class OrganisationsController {
...
+    const userId = req['user'].id;
+    const filePath = await this.organisationsService.exportOrganisationMembers(orgId, userId);
+    const fileStream = createReadStream(filePath);
+
+    res.set({
+      'Content-Type': 'text/csv',
+      'Content-Disposition': `attachment; filename="organisation-members-${orgId}.csv"`,
+    });
+
+    fileStream.pipe(res);
```
Comment (Cobidex) [inline src/modules/organisations/organisations.controller.ts:196]: Better to use pipeline to stream the files it offers better internal buffer management and reduces drainage. Your implementation is good, but pipeline is better for production

File: src/modules/organisations/organisations.controller.ts [hunk 2]
```diff
@@ -154,4 +156,51 @@ export class OrganisationsController {
...
+
+    fileStream.on('end', () => {
+      unlink(filePath, err => {
+        if (err) {
+          console.error('Failed to delete file:', err);
```
Comment (Cobidex) [inline src/modules/organisations/organisations.controller.ts:201]: You should use the built in nest logger to log your this at error level. Rather than console.error

File: src/modules/organisations/organisations.controller.ts [hunk 3]
```diff
@@ -154,4 +156,51 @@ export class OrganisationsController {
...
+        schema: {
+          type: 'string',
+          example: 'attachment; filename="organisation-members-{orgId}.csv"',
+        },
+      },
+    },
+  })
+  @UseGuards(OwnershipGuard)
+  @Get(':org_id/members/export')
+  async exportOrganisationMembers(@Param('org_id') orgId: string, @Req() req: Request, @Res() res: Response) {
```
Comment (Cobidex) [inline src/modules/organisations/organisations.controller.ts:186]: Add validation for UUID using the built in nest pipe for that. To ensure id is of proper type. This will prevent raising database validation exceptions

**Review by Cobidex — COMMENTED** [review]
It's no simple ticket well done..just fix comments

**Comment by kaleabendrias** [comment]
@Cobidex i used the pipeline, and removed the console.error with logger, and also i added parsing uuid.

**Review by Cobidex — APPROVED** [review]
